ocaml/xapi/xapi_session.ml

```diff
@@ -791,25 +791,38 @@ module Caching = struct
      disabled. This function exists to delay the construction of the
      cache, as Xapi_globs configuration is not guaranteed to have been
      populated before the top-level code of this module is executed. *)
-  let get_or_init_cache () =
-    if not !Xapi_globs.external_authentication_cache_enabled then
+  let get_or_init_cache ~__context =
+    (* read db here, Helpers.get_pool *)
+    let pool = Helpers.get_pool ~__context in
+    let cache_enabled =
+      Db.Pool.get_ext_auth_cache_enabled ~__context ~self:pool
+    in
+    if not cache_enabled then
       None
     else
-      let capacity = !Xapi_globs.external_authentication_cache_size in
+      let capacity =
+        Db.Pool.get_ext_auth_cache_size ~__context ~self:pool |> Int64.to_int
+      in
       let@ () = with_lock lock in
       match !cache with
       | Some _ as extant ->
           extant
       | _ ->
-          let auth_cache = AuthenticationCache.create ~size:capacity in
+          let expiry =
+            Db.Pool.get_ext_auth_cache_expiry ~__context ~self:pool
+            |> Int64.to_float
+            |> Clock.Timer.s_to_span
+            |> Option.value ~default:Mtime.Span.(300 * s)
```

Comment:
We can avoid using floats and some weirdness around precision and negative numbers:
```ocaml
            |> Int64.unsigned_to_int
            |> Option.map (fun sec -> Mtime.Span.(sec * s))
            |> Option.value ~default:Mtime.Span.(5 * m)
```